code/modules/clothing/modular_armor/tdf.dm

```diff
@@ -0,0 +1,359 @@
+		/obj/item/armor_module/module/mimir_environment_protection/mimir_helmet/mark1,
+		/obj/item/armor_module/module/welding,
+		/obj/item/armor_module/module/welding/superior,
+		/obj/item/armor_module/module/binoculars,
+		/obj/item/armor_module/module/binoculars/artemis_mark_two,
+		/obj/item/armor_module/module/artemis,
+		/obj/item/armor_module/module/antenna,
+		/obj/item/armor_module/module/night_vision,
+		/obj/item/armor_module/storage/helmet,
+		/obj/item/armor_module/armor/badge,
+	)
+	greyscale_config = null
+	starting_attachments = list(/obj/item/armor_module/storage/helmet)
+	icon_state_variants = list(
+		"black",
+	)
+	current_variant = "black"
+	visorless_offset_y = 0
+
+/obj/item/clothing/head/modular/tdf/hod
+	starting_attachments = list(/obj/item/armor_module/storage/helmet, /obj/item/armor_module/module/hod_head)
+
+/obj/item/clothing/head/modular/tdf/freyr
+	starting_attachments = list(/obj/item/armor_module/storage/helmet, /obj/item/armor_module/module/artemis)
+
+/obj/item/clothing/head/modular/tdf/antenna
+	starting_attachments = list(/obj/item/armor_module/storage/helmet, /obj/item/armor_module/module/antenna)
+
+/obj/item/clothing/head/modular/tdf/welding
+	starting_attachments = list(/obj/item/armor_module/storage/helmet, /obj/item/armor_module/module/welding)
+
+/obj/item/clothing/head/modular/tdf/superiorwelding
+	starting_attachments = list(/obj/item/armor_module/storage/helmet, /obj/item/armor_module/module/welding/superior)
+
+/obj/item/clothing/head/modular/tdf/mimir
+	starting_attachments = list(/obj/item/armor_module/storage/helmet, /obj/item/armor_module/module/mimir_environment_protection/mimir_helmet/mark1)
+
+/obj/item/clothing/head/modular/tdf/medium
+	name = "\improper Knight Class Helmet"
+	icon_state = "tdf_helmet_medium"
+	worn_icon_state = "tdf_helmet_medium"
+	soft_armor = list(MELEE = 50, BULLET = 70, LASER = 65, ENERGY = 65, BOMB = 55, BIO = 55, FIRE = 60, ACID = 55)
+
+/obj/item/clothing/head/modular/tdf/heavy
+	name = "\improper Bishop Class Helmet"
+	desc = "A bulky helmet paired with the 'Tyr' armor module, for added on additional protection at the cost of significant weight and reduced flexibility."
+	icon_state = "tdf_helmet_heavy"
+	worn_icon_state = "tdf_helmet_heavy"
+	soft_armor = list(MELEE = 60, BULLET = 80, LASER = 80, ENERGY = 80, BOMB = 65, BIO = 55, FIRE = 70, ACID = 60)
+
+/obj/item/clothing/head/modular/tdf/heavy/tyr
+	soft_armor = list(MELEE = 75, BULLET = 75, LASER = 75, ENERGY = 65, BOMB = 55, BIO = 55, FIRE = 55, ACID = 60)
+
+/obj/item/clothing/head/modular/tdf/pyro
+	name = "\improper Jester Class Helmet"
+	desc = "A bulky helmet with an integrated gas mask and a 'Surt' armor module for fireproofing, fielded to TDF flamethrower operators."
+	icon_state = "tdf_helmet_pyro"
+	worn_icon_state = "tdf_helmet_pyro"
+	soft_armor = list(MELEE = 45, BULLET = 65, LASER = 60, ENERGY = 60, BOMB = 50, BIO = 50, FIRE = 90, ACID = 50)
```

Comment:
the armour (and hat) used to have 100 fire armour, didn't check any other values.